client/src/featureform/resources.py

```diff
@@ -2278,9 +2280,27 @@ def __init__(
                 "The DatabricksCredentials requires only one credentials set ('username' and 'password' or 'host' and 'token' set.)"
             )
 
-        if not cluster_id:
+        if not self.cluster_id:
             raise Exception("Cluster_id of existing cluster must be provided")
 
+        if not self._validate_cluster_id():
+            raise ValueError(
+                "Cluster_id must follow the format: xxx-xxxxxx-xxxxxxxx, where x is alphanumeric"
+            )
+
+        if self.host and not self._validate_token():
+            raise ValueError(
+                "Token must start with 'dapi' followed by alphanumeric eg ex: dapixxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-x"
+            )
+
+    def _validate_cluster_id(self):
+        cluster_id_regex = r"^\w{3}-\w{6}-\w{8}$"
+        return re.match(cluster_id_regex, self.cluster_id)
+
+    def _validate_token(self):
+        token_regex = r"^dapi[\w-]+$"
```

Comment:
Should limit the number of characters. I believe the format I posted has the correct # of characters overall

Reply:
Got it! Just need a little clarification on the following:

Databricks example [here](https://www.databricks.com/blog/2022/08/03/how-to-build-a-marketing-analytics-solution-using-fivetran-and-dbt-on-the-databricks-lakehouse.html#:~:text=token%20(dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX)%3A) uses the following token format:
`dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`
Our's has a slight variation:
`dapixxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-x`
If the placement of hyphen always remains the same, here is the fix: a43176c4de6645b577761bc7f7986c8fc87faf99
If not, let me know.

Reply:
Yeah its interesting. Ours always have the dash in it

Reply:
Lets leave for now, we're adding some updates around this so we may add more variations later